### PR TITLE
[Merged by Bors] - refactor: fix LinearMap.coe_mk

### PR DESCRIPTION
**Pull request by Komyyy**
---

This PR fix `LinearMap.coe_mk`.
```
@[simp]
theorem coe_mk {σ : R →+* S} (f : M →+ M₃) (h) :
    ((LinearMap.mk f h : M →ₛₗ[σ] M₃) : M → M₃) = f
```
to
```
@[simp]
theorem coe_mk {σ : R →+* S} (f : M → M₃) (h₁ h₂) :
    ((LinearMap.mk (AddHom.mk f h₁) h₂ : M →ₛₗ[σ] M₃) : M → M₃) = f
```

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by ChrisHughes24** [comment]
This seems like it makes the lemma less general. I would suggest maybe changing the statement of `LinearMap.coe_mk` to the following (changing the coercion from `M → M₃` to `M →+ M₃`.
```lean
theorem coe_mk {σ : R →+* S} (f : M →+ M₃) (h) :
    ((LinearMap.mk f h : M →ₛₗ[σ] M₃) : M →+ M₃) = f
```
You could also add `AddHom.coe_mk` if it's not there already and then the combination of the two lemmas should be able to prove you version of `LinearMap.coe_mk`.

What do you think @eric-wieser ?

**Comment by eric-wieser** [comment]
We should have both the version with coe and with coe_fn; I'm a little surprised to hear that Lean3 only has the second of these. Maybe it has both and we forgot to port one of them

**Comment by Komyyy** [comment]
`AddHom.coe_mk` is already in the library with a `simp` attribute, but `simp` couldn't prove my first version of `LinearMap.coe_mk`.
I felt something is wrong, and finally found that `→+` is actually not `AddHom` but `AddMonoidHom`. 🤯
So, the best way is this.
```
@[simp]
theorem coe_mk {σ : R →+* S} (f : AddHom M M₃) (h) :
    ((LinearMap.mk f h : M →ₛₗ[σ] M₃) : M → M₃) = f
```
I also add this.
```
@[simp]
theorem coe_mk' {σ : R →+* S} (f : AddHom M M₃) (h) :
    ((LinearMap.mk f h : M →ₛₗ[σ] M₃) : AddHom M M₃) = f
```

File: Mathlib/Algebra/Module/LinearMap.lean [hunk 1]
```diff
@@ -269,11 +269,17 @@ protected def Simps.apply {R S : Type _} [Semiring R] [Semiring S] (σ : R →+*
...
 @[simp]
-theorem coe_mk {σ : R →+* S} (f : M →+ M₃) (h) :
+theorem coe_mk {σ : R →+* S} (f : AddHom M M₃) (h) :
     ((LinearMap.mk f h : M →ₛₗ[σ] M₃) : M → M₃) = f :=
   rfl
 #align linear_map.coe_mk LinearMap.coe_mk
 
+-- Porting note: This theorem is added.
+@[simp]
+theorem coe_mk' {σ : R →+* S} (f : AddHom M M₃) (h) :
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/LinearMap.lean:279]: perhaps
```suggestion
theorem coe_addHom_mk {σ : R →+* S} (f : AddHom M M₃) (h) :
```

File: Mathlib/Algebra/Module/LinearMap.lean [hunk 2]
```diff
@@ -269,11 +269,17 @@ protected def Simps.apply {R S : Type _} [Semiring R] [Semiring S] (σ : R →+*
 initialize_simps_projections LinearMap (toAddHom_toFun → apply)
 
 @[simp]
-theorem coe_mk {σ : R →+* S} (f : M →+ M₃) (h) :
+theorem coe_mk {σ : R →+* S} (f : AddHom M M₃) (h) :
     ((LinearMap.mk f h : M →ₛₗ[σ] M₃) : M → M₃) = f :=
   rfl
 #align linear_map.coe_mk LinearMap.coe_mk
 
+-- Porting note: This theorem is added.
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/LinearMap.lean:277]:
```suggestion
-- Porting note: This theorem is new.
```

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Comment by jcommelin** [comment]
bors areyouasleep?

bors merge

**Pull request closed without merging**